File: LeetCode_POTD/July_2026/July_9.java
package Leetcode_POTD.July_2026;

/*
Divide in groups (partiotion from point where differnce in neighbours is more than maxDiff)
If queries point in same group return yes.

TC - O(n+m), SC - O(n)
*/

public class July_9 {
    public boolean[] pathExistenceQueries(int n, int[] nums, int maxDiff, int[][] queries) {
        int a[]=new int[n];
        int grp=1;
        a[0]=1;
        for(int i=0;i<n-1;i++){
            if(nums[i+1]-nums[i]<=maxDiff) a[i+1]=grp;
            else a[i+1]=++grp;
        }
        boolean ans[]=new boolean[queries.length];
        for(int i=0;i<queries.length;i++){
            if(a[queries[i][0]]==a[queries[i][1]]) ans[i]=true;
        }
        return ans;
    }
}

File: LeetCode_POTD/July_2026/July_9.py
"""
Divide in groups (partiotion from point where differnce in neighbours is more than maxDiff)
If queries point in same group return yes.

TC - O(n+m), SC - O(n)
"""

class Solution:
    def pathExistenceQueries(self, n: int, nums: List[int], maxDiff: int, queries: List[List[int]]) -> List[bool]:
        a=[]
        grp=1
        a.append(1)
        for i in range(n-1):
            if nums[i+1]-nums[i]>maxDiff: grp+=1
            a.append(grp)
        ans=[]
        for u,v in queries:
            if(a[u]==a[v]): ans.append(True)
            else: ans.append(False)
        return ans